src/Compilers/CSharp/Portable/Symbols/UnboundGenericType.cs

```diff
@@ -119,4 +119,75 @@ public override int GetHashCode()
                 : Hash.Combine(_name, _errorInfo.Code);
         }
     }
+
+    /// <summary>
+    /// Used for lightweight binding of type constraints. Instead of binding type arguments,
+    /// we'll just use these placeholders instead. That's good enough binding to compute
+    /// <see cref="TypeSymbol.IsValueType"/> with minimal binding.
+    /// </summary>
+    internal sealed class PlaceholderTypeArgumentSymbol : ErrorTypeSymbol
```

Comment:
>PlaceholderTypeArgumentSymbol [](start = 26, length = 29)

Could we use `UnboundArgumentErrorTypeSymbol` instead?

Reply:
We could (and I did initially), but I felt that seeing a distinct type would be helpful to someone who ever sees this when debugging. It's clearly its own thing (I think having a distinct type is good).
I could try and share more code though...